File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // get text from user
    string text = get_string("Text: ");
    
    // letter count
    float count_letters = 0;
    // word count, start at 1 to account for first word
    float count_words = 1;
    // sentence count
    float count_sentences = 0;
    
    // go through each letter
    for (int i = 0; i < strlen(text); i++)
    {
        // count only alphabet characters
        if (isalpha(text[i]))
        {
            count_letters++;
        }
        
        // check for spaces
        if (isspace(text[i]))
        {
            count_words++;
        }
        
        // check for punctuation
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count_sentences++;
        }

    }
    
    //printf("%f letter(s)\n", count_letters);
    //printf("%f words(s)\n", count_words);
    //printf("%f sentences(s)\n", count_sentences);
    
    float L = (count_letters / count_words) * 100;
    float S = (count_sentences / count_words) * 100;  
    
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    if (index > 16)
    {
        printf("Grade 16+\n"); 
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n"); 
    }
    else
    {
        printf("Grade %i\n", index);
    }

}

File: readability.py
def main():
    # get text to analyze
    text = input("Text: ")
    
    # letter count
    count_letters = 0
    # word count, start at 1 to account for first word
    count_words = 1
    # sentence count
    count_sentences = 0
    
    # go through each letter
    for letter in text:
        # count only alphabet characters
        if letter.isalpha():
            count_letters += 1
        
        # check for spaces
        if (letter == " "):
            count_words += 1

        # check for punctuation
        if letter == '.' or letter == '!' or letter == '?':
            count_sentences += 1
    
    L = (count_letters / count_words) * 100
    S = (count_sentences / count_words) * 100  
    
    index = int(round(0.0588 * L - 0.296 * S - 15.8))
    
    # output grade 
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1") 
    else:
        print("Grade " + str(index))
   
    
# execute main function
main()
